plrust/src/user_crate/cargo.rs

```diff
@@ -0,0 +1,158 @@
+                    std::env::join_paths(vec![path.as_path(), std::path::Path::new("/usr/bin")])?,
+                );
+            } else {
+                // we don't have a home directory... where could cargo be?  Ubuntu installed cargo
+                // at least puts it in /usr/bin
+                command.env("PATH", "/usr/bin");
+            }
+        }
+    }
+    Ok(())
+}
+
+/// PL/Rust uses is own rustc driver named `plrustc`, and it is expected that it be on the path.
+/// We use our own rustc driver so we can enable our various lints
+fn configure_rustc(command: &mut Command) {
+    // TODO:  Do we want a GUC to set this if maybe `which plrustc` fails?
+    command.env("RUSTC", "plrustc");
+}
+
+/// There was a time in the past where plrust had a `plrust.pg_config` GUC whose value was
+/// passed down to the "pgx-pg-sys" transient dependency via an environment variable.
+///
+/// This turned out to be an unwanted bit of user, system, and operational complexity.
+///
+/// Instead, we tell the environment that "pg_config" is described as environment
+/// variables, and set every property Postgres can tell us (which is essentially how
+/// `pg_config` itself works) as individual environment variables, each prefixed with
+/// "PGX_PG_CONFIG_".
+///
+/// "pgx-pg-sys"'s build.rs knows how to interpret these environment variables to get what
+/// it needs to properly generate bindings.
+fn configure_pg_config(
+    command: &mut Command,
+    cross_compilation_target: Option<CrossCompilationTarget>,
+) {
+    command.env("PGX_PG_CONFIG_AS_ENV", "true");
+    for (k, v) in pg_config_values() {
+        let k = format!("PGX_PG_CONFIG_{k}");
+        command.env(k, v);
+    }
+
+    // set environment variables we need in order for a cross compile
+    if let Some(target_triple) = cross_compilation_target {
+        // the CARGO_TARGET_xx_LINKER variable
+        let (k, v) = target_triple.linker_envar();
+        command.env(k, v);
+
+        // pgx-specified variable for where the bindings are
+        if let Some((k, v)) = target_triple.bindings_envar() {
+            command.env(k, v);
+        }
+    }
+}
+
+/// these are environment variables that could possibly impact pgx compilation that we don't
+/// want to accidentally or purposely be inherited from the running "postgres" process
+fn sanitize_env(command: &mut Command) {
+    command.env_remove("DOCS_RS"); // we'll never be building user function on https://docs.rs
+    command.env_remove("PGX_BUILD_VERBOSE"); // unnecessary to ever build a user function in verbose mode
+    command.env_remove("PGX_PG_SYS_GENERATE_BINDINGS_FOR_RELEASE"); // while an interesting idea, PL/Rust user functions are not used to generate a `pgx` release
+    command.env_remove("CARGO_MANIFEST_DIR"); // we are in the manifest directory b/c of `command.current_dir()` above
+    command.env_remove("OUT_DIR"); // rust's default decision for OUT_DIR is perfectly acceptable to PL/Rust
```

Comment:
...Mm. Are we sure? Previously we had deliberately set `OUT_DIR` so that multiple compilations could reuse artifacts when they were cached.

Reply:
PL/Rust has never set `OUT_DIR`.  We do set `CARGO_TARGET_DIR`, which I think is what you're thinking of.

Reply:
oh, duh.